src/app/[locale]/page.tsx

```diff
@@ -1,15 +1,23 @@
 import {PageWithSidebarContainer} from '@/components/page';
 import datasetFetcher from '@/lib/dataset-fetcher-instance';
-import {useLocale} from 'next-intl';
+import {useLocale, NextIntlClientProvider} from 'next-intl';
 import DatasetList from './dataset-list';
 
 export default async function Home() {
   const initialSearchResult = await datasetFetcher.search();
   const locale = useLocale();
-
+  const messages = (await import(`@/messages/${locale}.json`)).default;
   return (
     <PageWithSidebarContainer>
-      <DatasetList initialSearchResult={initialSearchResult} locale={locale} />
+      <NextIntlClientProvider
+        locale={locale}
+        messages={{Home: messages.Home, Paginator: messages.Paginator}}
```

Comment:
Do we have to be specific about the messages that we want to use ('Home', 'Paginator', etc.)? Or is it possible to pass in the full `messages` instead (and let downstream components figure out which messages they need)?

Reply:
This is different for server and client components.

For the server components, all the messages are loaded with the loader in next.config.js. And can be used everywhere.

For client components, the advice is to only use NextIntlClientProvider where needed, to keep the bundle size as small as possible. So also only send the messages needed, for the smallest bundle size. 

From the docs: https://next-intl-docs.vercel.app/docs/next-13/server-components#switching-to-client-components

" If you absolutely need to use functionality from next-intl on the client side, you can wrap the respective components with NextIntlClientProvider ([example code](https://github.com/amannn/next-intl/blob/feat/next-13-rsc/packages/example-next-13-advanced/src/components/client/02-MessagesOnClientCounter/Counter.tsx)). Note however that this will increase your client bundle size. "

Reply:
Thanks, that makes sense!

My hope was that next-intl would do some tree shaking when a production build of the app is created, by only keeping the messages that are being used client-side.